src/runtime_src/core/tools/common/XBHelpMenusCore.cpp

```diff
@@ -557,3 +558,53 @@ XBUtilities::report_subcommand_help( const std::string &_executableName,
       std::cout << fmtExtHelp % formattedString;
   }
 }
+
+std::vector<std::string>
+XBUtilities::process_arguments( po::variables_map& vm,
+                                po::command_line_parser& parser,
+                                const po::options_description& options,
+                                const po::positional_options_description& positionals,
+                                bool validate_arguments
+                                )
+{
+  // Add unregistered "option"" that will catch all extra positional arguments
+  po::options_description _options(options);
+  _options.add_options()("__unreg", po::value<std::vector<std::string> >(), "Holds all unregistered options");
+  po::positional_options_description _positionals(positionals);
+  _positionals.add("__unreg", -1);
+
+  // Parse the given options and hold onto the results
+  auto parsed_options = parser.options(_options).positional(_positionals).allow_unregistered().run();
+  
+  if (validate_arguments) {
+    // This variables holds options denoted with a '-' or '--' that were not registered
+    auto unrecognized_options = po::collect_unrecognized(parsed_options.options, po::exclude_positional);
+    // Parse out all extra positional arguments from the boost results
+    // This variable holds arguments that do not have a '-' or '--' that did not have a registered positional space
+    std::vector<std::string> extra_positionals;
+    for (auto option : parsed_options.options)

```

Comment:
Great use of not using curly braces.   Being that there are multiple level here, you might wish to consider adding a set for the outer most "set".   This will allow future developers recongizise the pattern.

Reply:
While not the most important performance issue we have, perhaps some `const auto&` in all these range-for loops when it makes sense?